src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -117,12 +175,16 @@ public override void VisitRazorDirectiveBody(RazorDirectiveBodySyntax node)
             {
                 AddSemanticRange(node.Keyword, RazorSemanticTokensLegend.RazorDirective);
             }
-            base.VisitRazorDirectiveBody(node);
         }
 
         public override void VisitMarkupTagHelperStartTag(MarkupTagHelperStartTagSyntax node)
         {
             AddSemanticRange(node.OpenAngle, RazorSemanticTokensLegend.MarkupTagDelimiter);
+            if (node.Bang != null)
```

Comment:
Ah this was a separate fix right?

Reply:
Kinda? I was just making sure we visited all child nodes, but this is certainly more correct.